### ISSUE TITLE
Implement Single Sign On (SSO) for Windows Enterprise Users

### ISSUE THREAD

**Issue by deleolajide** [igniterealtime/Pade#66]
Use the Waffle library to implement SSO for Windows Active Directory Users
See https://github.com/Waffle/waffle

**Comment by deleolajide** [comment]
This issue is re-opened in response to a request at the Ignite Realtime weekly meeting to support corporate deployment of Pàdé with Windows SSO to many employees in a single simple effort.

The issues identified are:

- Take advantage of the group policy deployment tools for Chrome Extensions provided by Google.
- Pre-configure Pàdé with connection settings
- Streamline Pàdé options and settings to corporate standards.

A possible solution with branding was identified in https://github.com/igniterealtime/Pade/issues/89

However, this does not cover the majority use case of using the stock [Pade@Work](https://chrome.google.com/webstore/detail/pade-unified-communicatio/fohfnhgabmicpkjcpjpjongpijcffaba) version available from the Chrome Web Store.

**Comment by deleolajide** [comment]
In order to perform a *zero-install* of Pade@Work with Windows SSO will require the following steps

- Create a branding file (branding.json) and copy to the root folder of your Openfire web server (OPENFIRE_HOME/resources/spank). . Make sure winSSO is enabled. See https://github.com/igniterealtime/Pade/issues/89 for details
- Send out an email to all users involved informing them to keep the Openfire Meetings home page opened in their browsers. Include the actual link for them to click on.
- Perform a [group policy deployment with the tools for Chrome Extensions](http://dennisspan.com/deploying-google-chrome-extensions-using-group-policy/) provided by Google.
- If you include the link to Pade@Work in the email and ask them to self-install, then the group policy deployment step can be skipped.
- When Pade@Work gets installed, it will check for the opened Openfire Meetings web page in the browser. If found, it will then extract the server name and port to configure Pade@Work and then do a restart. If Windows SSO was enabled and user account correctly setup in Openfire, the user should be automatically logged in.

**Comment by deleolajide** [comment]
This is now implemented in version 0.8.0

You would need a file called pade.json in OPENFIRE_HOME/resources/spank. The minimum parameters are
```
{
    "useWebsocket"    : {"disable": false, "value": true},
    "useWinSSO"       : {"disable": false, "value": true}
} 
```
 Make sure Openfire Meetings is already opened in any browser tab. 

![image](https://user-images.githubusercontent.com/110731/47997125-b51bb700-e0f2-11e8-91dd-dc96a60e96c2.png)

If not and you get the Pàdé options/login page, simply re-install Pàdé after opening the page. If all goes well, Pàdé should be auto-configured with the same openfire server as Openfire Meetings and the user configured as the Windows desktop user.

![image](https://user-images.githubusercontent.com/110731/47997288-36734980-e0f3-11e8-8a5a-1bbe1413b91f.png)
